### DISCUSSION TITLE
Changing Primary Color

### DISCUSSION THREAD

**Discussion by hatemabsi** [retypeapp/retype#282] [Q&A] [answered]
Hello,

I've been working on Retype for a while now and it has been working perfectly! I was wondering if there was a way or if there is an intention to add a way to change the primary (blue) color into different colors?

Examples:
![image](https://user-images.githubusercontent.com/84775981/156431484-4bca0fb6-45da-41bb-b445-41f8d8911a13.png)
![image](https://user-images.githubusercontent.com/84775981/156431519-524a71ce-7567-4c02-bc73-0a30a216f945.png)
![image](https://user-images.githubusercontent.com/84775981/156431559-3ae15b7f-3856-4a21-ab30-740246881fcd.png)

Example of what it would look like if the primary color was changed to red!

![image](https://user-images.githubusercontent.com/84775981/156432207-ac518b70-b7ae-485b-8f42-7f4ad1b89f59.png)

**Comment by fabriciomurta** [comment]
Yes, there is the intention and some requests already, you might want to subscribe to the issue #42 if you want to be notified when it is implemented.

Discussion #127 is also on this topic.

There's also #159, #81, #57 related to this so, a well requested feature.

**Comment by geoffreymcgill** [comment, marked as answer]
Hi @PrinceHatem. We were hoping to get some palette configuration options built into the upcoming Retype `v2.0` release, but they are going to have to wait for a future release. It ended up being more complicated than we had initially budgeted the time for. 

That said, we have added some hooks so that you will be able to manually add (or override) css styles. You should be able to change the colors with css overrides as of `v2.0`. 

Hope this helps.

**Reply by hatemabsi** [reply]
Awesome! The css overrides would work perfectly! Do we have a timeline of when to expect v2.0?

**Reply by geoffreymcgill** [reply]
> Do we have a timeline of when to expect v2.0?

The Retype `v2.0.0-preview` release is available now. Please see https://github.com/retypeapp/retype/discussions/287#discussioncomment-2321442.

As of `v2.0`, you can add the file `/_includes/head.html` and any of contents of that .html file will be included in all pages of the Retype generated website. 

The file must be called `head.html` and must be located within a root `/_includes/` folder. Only raw html is supported. 

You could create a custom .css file and include in all pages of your website. For example, create a `/static/custom.css` file and then within your `_includes/head.html` file, add a link to that `custom.css` file. The file will then be loaded on all pages of the website.

```html
<link href="/static/custom.css" rel="stylesheet" />
```

This `head.html` is part of a list of new magic include files. Other options include:

Name | Description
--- | ---
`_includes/head.html` | The contents of the `head.html` file are inserted at the bottom of the `<head>` section, just before the closing `</head>` tag.
`_includes/head-top.html` |  The contents of the `head-top.html` file are inserted at the top of the `<head>` section, just after the opening `<head>` tag.
`_includes/body.html` |  The contents of the `body.html` file are inserted at the bottom of the `<body>` section, just before the closing `</body>` tag.
`_includes/body-top.html` |  The contents of the `body-top.html` file are inserted at the top of the `<body>` section, just after the opening `<body>` tag.
`_includes/top.md` |  A custom Markdown file to be included as the common top content on all generated pages of the website.
`_includes/bottom.md` |  A custom Markdown file to be included as the common bottom content on all generated pages of the website.

Hope this helps.

**Reply by hatemabsi** [reply]
Thank you very much!

**Reply by simonokeefe** [reply]
With regards to the initial question, how did you end up changing the primary color? I understand it involves adding something in `_includes/head.html`. What is the setting required to change the primary color? Thanks!

**Reply by rdewit** [reply]
The following page shows how to set a custom style using `_includes/head.html`: https://retype.com/components/container/#custom-global-css

**Reply by DannyBen** [reply]
This is a good table. Shouldn't it be a part of the documentation?

**Reply by woigl** [reply]
This is a great feature, but could you enhance that the path in the head.html file to the CSS file will use the URL setting from the project file?

Would you like me to create a new issue for this?

**Reply by geoffreymcgill** [reply]
> This is a great feature, but could you enhance that the path in the head.html file to the CSS file will use the URL setting from the project file?

Can you explain this further? I'm not 100% sure what you mean.

**Reply by woigl** [reply]
The recommendation is to put the CSS import into the `_includes/head.html` file., which I did, and it works fine if I set an absolute path for the CSS location. The issue is that my documentation is not placed into the root path.

That is the content you recommend:
```html
<link href="/static/custom.css" rel="stylesheet" />
```

That is what I would like to do:
```html
<link href="{{PROJECT.url}}/static/custom.css" rel="stylesheet" />
```

Which should resolve into something like this:
```html
<link href="https://example.com/somepath/static/custom.css" rel="stylesheet" />
```

The `PROJECT.url` variable is what I set in the project file at the key `url`.

**Reply by geoffreymcgill** [reply]
Try using the following path in the `_includes/head.html`:

```html
<link href="./static/custom.css" rel="stylesheet" />
```

**Reply by woigl** [reply]
Unfortunately, that does not work. It resolves to the wrong path.

**Reply by geoffreymcgill** [reply]
Can you start a new thread with this [issue](https://github.com/retypeapp/retype/issues)?
